KsApi/MockService.swift

```diff
@@ -361,6 +363,7 @@ internal struct MockService: ServiceType {
     self.submitApplePayResponse = submitApplePayResponse
 
     self.toggleStarResponse = toggleStarResponse
+    self.toggleStarError = toggleStarError
```

Comment:
It looks like you haven't tied this error to the function where it is being called. If you look at the `toggleStar(_ project:)` method, you can see that if `toggleStarResponse` is nil, it returns the same error every time: `.init(error: .couldNotParseJSON)`. Instead, you'll want to update the method to use your `toggleStarError`.